TESTS/network/wifi/wifi_connect_nocredentials.cpp

```diff
@@ -26,7 +26,9 @@ using namespace utest::v1;
 void wifi_connect_nocredentials(void)
 {
     WiFiInterface *wifi = get_interface();
-    nsapi_error_t error;
-    error = wifi->connect();
-    TEST_ASSERT(error == NSAPI_ERROR_NO_SSID || error == NSAPI_ERROR_PARAMETER);
+    nsapi_error_t error_connect, error_disconnect;
+    error_connect = wifi->connect();
+    error_disconnect = wifi->disconnect();
+    TEST_ASSERT(error_connect == NSAPI_ERROR_NO_SSID || error_connect == NSAPI_ERROR_PARAMETER);
+    TEST_ASSERT(error_disconnect == NSAPI_ERROR_NO_CONNECTION);
```

Comment:
Hi
Back to this patch...
Why the disconnect function has to return NSAPI_ERROR_NO_CONNECTION, and not NSAPI_ERROR_OK ?

Reply:
In this particular case, there is no connection at all when the disconnect is called.